bin/updateSourceModeration.ts

```diff
@@ -0,0 +1,32 @@
+import '../src/config';
+import createOrGetConnection from '../src/db';
+import { SquadSource } from '../src/entity';
+import { runInQueueStream } from './common';
+
+interface Result {
+  title: string;
+  id: string;
+}
+
+(async () => {
+  const con = await createOrGetConnection();
+  await con.transaction(async (manager) => {
+    console.log('initializing stream');
+
+    const builder = manager
+      .getRepository(SquadSource)
+      .createQueryBuilder('ss')
+      .select('ss.id', 'id')
+      .where('ss."memberPostingRank" = 0');
+
+    const stream = await builder.stream();
+
+    await runInQueueStream<Result>(stream, async ({ id }: Result) => {
+      await manager
+        .getRepository(SquadSource)
+        .update({ id }, { moderationRequired: true });
+    });
```

Comment:
@sshanzel can't this be done in a single query?

```sql
update source s set "moderationRequired" = true 
where s.type = 'squad' and s."memberPostingRank" = 0
```

Reply:
if yes we can even just run it as once off on the DB, no need to create a script

Reply:
It can, like most of our updates. If I remember correctly, we run these scripts to manage the amount of updates per batch.

If we don't have to worry the amount this time, then yeah, we can just execute this as a regular sql script.

Reply:
you can check how many squads it takes, also if you want to update per batch usually we use limit/offset with `insert from select` or date based.

@sshanzel its only 8k rows `select count(1) from source s where s.type = 'squad' and s."memberPostingRank" = 0`

so you can do it as a one time query

Reply:
You can check how many squads it takes, also if you want to update per batch usually we use limit/offset.

@sshanzel its the small amount of rows, see `select count(1) from source s where s.type = 'squad' and s."memberPostingRank" = 0`

so you can do it as a one time query 🚀

Reply:
We've got 8k rows. Wdyt? Do we run it in batches?

![Screenshot 2024-11-11 at 9 42 57 PM](https://github.com/user-attachments/assets/f1e594de-1c1f-4b28-b0ca-c0a0ddcff328)